internal/build/custom_builder.go

```diff
@@ -184,9 +185,20 @@ func (b *CustomBuilder) readImageRef(ctx context.Context, outputsImageRefTo stri
 			outputsImageRefTo, err)
 	}
 
-	// TODO(nick): Add support for separate local and cluster refs.
+	clusterRef := ref
+	if reg != nil && reg.HostFromContainerRuntime != "" {
+		replacedName, err := container.ParseNamed(strings.Replace(ref.Name(), reg.Host, reg.HostFromContainerRuntime, 1))
```

Comment:
Is there a better way to do this?

Reply:
There is! It's not as discoverable since I flattened the Registry types recently, but it's here: https://github.com/tilt-dev/tilt/blob/33fdb81f7bbbf240edb35d17103f7b3936c39244/internal/container/registry.go#L111-L123

(And for completeness, to get the ref type it wants -> `container.NewRefSelector(ref)`)

Reply:
Hmm, I think I tried this at one point. It would turn a ref like `localhost:500/foo-bar` into `remote:5000/localhost_5000_foo_bar`, which isn't the same since the custom builder is pushing the image to the registry.

Reply:
Ahhh d'oh, you're totally right!

Reply:
Cool, thanks! Just wanted to make sure I didn't miss anything else.